src/utl/include/utl/Logger.h

```diff
@@ -35,6 +35,9 @@
 
 #pragma once
 
+// disable the newline in spdlog and handle via pattern, so partial lines can be printed
+#define SPDLOG_EOL ""
```

Comment:
Is this hack why you are using the header only version?

Reply:
Yes, this is how the spdlog suggests removing the new line; however, this is not the only way (it is just the one that resulted in the fewest number of changes). It's also possible to use `set_formatter` (instead of `set_pattern`) and instantiate a formatter without newlines. I think that would allow the compile with `spdlog::spdlog` and not require this directive. When I tried earlier, I has issues getting it to pass on clang7 (and since this worked I didn't press forward on that path), but I'm happy to try that approach if you think it would be better?